src/esp/assets/PTexMeshData.cpp

```diff
@@ -582,55 +660,77 @@ void PTexMeshData::uploadBuffersToGPU(bool forceReload) {
     currentMesh->ibo.setData(submeshes_[iMesh].ibo,
```

Comment:
@mosra:
I am debugging the segfault.
After this "setData", do you think currentMesh->ibo.size() would equal to submeshes_[iMesh].ibo.size()?
Same for the vbo.

Reply:
I am asking this question is because I observed the size of currentMesh->ibo.size() was 4x of the submeshes_[iMesh].ibo.size(). 
I suspect currentMesh->ibo.size() returns the bytes of the entire array.
If this is the case, later "setCount(currentMesh->ibo.size())" would be wrong.

Reply:
> After this "setData", do you think currentMesh->ibo.size() would equal to submeshes_[iMesh].ibo.size()?

The naming is making this harder than it should be, first I assumed the two `ibo`s are both `std::vector`s but they're actually not :) Calling [GL::Buffer::size()](https://doc.magnum.graphics/magnum/classMagnum_1_1GL_1_1Buffer.html#a5e9a38c6e0d1e3e818d8648c0ac141b7) returns a number of bytes, since the buffer is a typeless piece of memory. And it's a query into GL, which is best avoided (moreover this function is not available at all in OpenGL ES or WebGL, but I'm not sure if that matters here) -- if you have the same information available CPU-side, it's recommended to prefer using that -- the `size()`, `data()`, `imageSize()`, ... functions on GL objects are meant mainly for debugging purposes.

Few lines below in the diff I see this -- could it be related?

```diff
-        .setCount(currentMesh->ibo.size() / 2)
+        .setCount(currentMesh->ibo.size())
```

Reply:
Yes, that is the problem. We actually need to pass the "number of primitives" to the setCount(), not the "number of bytes". That is why no matter if we divide it by 2, it will segfault anyway.

Reply:
Do you have any other example in your docs that shows how to correctly use ibo, vbo?
The "textured triangle" demo is a bit too simple in this case, in my opinion.

Reply:
The Primitives example expands on that, and the GL::Mesh documentation is very extensive.